dask/array/gufunc.py

```diff
@@ -602,6 +616,7 @@ def __init__(self, pyfunc, **kwargs):
         self.output_sizes = kwargs.pop("output_sizes", None)
         self.output_dtypes = kwargs.pop("output_dtypes", None)
         self.allow_rechunk = kwargs.pop("allow_rechunk", False)
+        self.meta = kwargs.pop("meta", None)
```

Comment:
Should we check, and throw a warning in case of, mismatch between `output_dtypes` and the dtypes as given in `meta`?